chemprop/cli/train.py

```diff
@@ -467,6 +475,9 @@ def add_train_args(parser: ArgumentParser) -> ArgumentParser:
 
 
 def process_train_args(args: Namespace) -> Namespace:
+    if args.dry_run:
+        return args
+
```

Comment:
Why do you skip `process_train_args` if it is a dry run? It looks like this function just raises errors if the user gives commands that don't make sense. I think we want to keep these checks during a dry run.

Reply:
Originally it was because I thought no config_path was needed, but on second glance it seems that anyways a data_path will be provided so there should be no errors raised. Thanks for the catch.